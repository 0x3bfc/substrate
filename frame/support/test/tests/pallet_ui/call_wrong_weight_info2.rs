#[frame_support::pallet]
mod pallet {
	use frame_support::pallet_prelude::{Hooks, DispatchResultWithPostInfo};
	use frame_system::pallet_prelude::{BlockNumberFor, OriginFor};

	#[pallet::config]
	pub trait Config: frame_system::Config {}

	#[pallet::pallet]
	pub struct Pallet<T>(core::marker::PhantomData<T>);

	#[pallet::hooks]
	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}

	struct NotWeight;
	struct WeightInfo;
	impl WeightInfo {
		fn foo() -> NotWeight {
			unimplemented!();
		}
	}
	#[pallet::call]
	#[pallet::weight_info(WeightInfo)]
	impl<T: Config> Pallet<T> {
		fn foo(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
			unimplemented!();
		}
	}
}

fn main() {
}
