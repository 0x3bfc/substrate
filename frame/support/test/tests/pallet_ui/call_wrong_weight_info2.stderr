error[E0277]: the trait bound `NotWeight: WeighData<()>` is not satisfied
  --> $DIR/call_wrong_weight_info2.rs:22:12
   |
22 |     #[pallet::call]
   |               ^^^^ the trait `WeighData<()>` is not implemented for `NotWeight`
   |
   = note: required for the cast to the object type `dyn WeighData<()>`

error[E0277]: the trait bound `NotWeight: ClassifyDispatch<()>` is not satisfied
  --> $DIR/call_wrong_weight_info2.rs:22:12
   |
22 |     #[pallet::call]
   |               ^^^^ the trait `ClassifyDispatch<()>` is not implemented for `NotWeight`
   |
   = note: required for the cast to the object type `dyn ClassifyDispatch<()>`

error[E0277]: the trait bound `NotWeight: PaysFee<()>` is not satisfied
  --> $DIR/call_wrong_weight_info2.rs:22:12
   |
22 |     #[pallet::call]
   |               ^^^^ the trait `PaysFee<()>` is not implemented for `NotWeight`
   |
   = note: required for the cast to the object type `dyn PaysFee<()>`
